flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/CommitterOperator.java

```diff
@@ -181,7 +181,7 @@ private void commitAndEmitCheckpoints() throws IOException, InterruptedException
     private void commitAndEmit(CheckpointCommittableManager<CommT> committableManager)
             throws IOException, InterruptedException {
         Collection<CommittableWithLineage<CommT>> committed = committableManager.commit(committer);
```

Comment:
NIT: as we are no longer using committed in the if, we could move `committableManager.commit(committer)` inlined to line 189 and remove the `committed` variable

Reply:
`commit` is unfortunately not side-effect free; we would never proceed with your proposal. But maybe it should be side-effect free. 

However, I'd like to pick that idea up on master and not the backport (this PR).